File: MessagingApp-ApiContext/Models/Message.cs
using System;
using System.Collections.Generic;

namespace MessagingApp.ApiContext.Models
{
    public class Message
    {
        public int Id { get; set; }

        public string MessageText { get; set; }

        public int? UserId { get; set; }
        public User User { get; set; }

        public int? ConversationId { get; set; }
        public Conversation Conversation { get; set; }

        public ICollection<MessageReaction> MessageReactions { get; set; }

        public DateTime Created { get; set; }

        public DateTime LastUpdated { get; set; }
    }
}

File: MessagingApp-App/ClientApp/src/app/_models/message.ts
import { Conversation } from "./conversation";
import { MessageReaction } from "./messageReaction";
import { User } from "./user";

export interface Message {
  id?: number;
  messageText: string;
  userId?: number;
  user: User;
  conversationId?: number;
  conversation: Conversation;
  messageReactions?: MessageReaction[];
  created?: Date;
  lastUpdated?: Date;
}
